File: llm_generated_code/ChatGPT/Python/Level1/Q28.py
def convert_case(word):
    lower_count = sum(1 for c in word if c.islower())
    upper_count = sum(1 for c in word if c.isupper())
    return word.lower() if lower_count >= upper_count else word.upper()

word = input()
result = convert_case(word)
print(result)

File: llm_generated_code/ChatGPT/CPP/Level1/Q28.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    string word;
    cin >> word;
    int lowerCount = 0, upperCount = 0;

    for (char c : word) {
        if (islower(c)) lowerCount++;
        else upperCount++;
    }

    cout << (lowerCount >= upperCount ? 
              static_cast<string>(word.begin(), word.end()).append(1, tolower(word[0])) : 
              static_cast<string>(word.begin(), word.end()).append(1, toupper(word[0]))) << endl;

    for (char& c : word) {
        c = (lowerCount >= upperCount) ? tolower(c) : toupper(c);
    }
    
    cout << word << endl;
    return 0;
}
